### PR TITLE
8087557: [Win] [Accessibility, Dialogs] Alert Dialog content is not fully read by Screen Reader

### PR DESCRIPTION
**Pull request by arapte**
Accessibility is not implemented for JavaFX dialogs. This change is to implement the accessibility on windows platform.
Without this fix : On Windows platform, content of Dialog are not read by Windows Narrator or JAWS.
With this fix:
1. Windows narrator reads the dialog correctly
2. JAWS fails to read the header text. : This will be handled separately 
3. Behavior of VoiceOver on MacOS remains unchanged : This will be handled separately.

Verification:
1. Run HelloAlert toy app, with fix.
2. Start Windows Narrator
3. Click on the different buttons to show different Dialogs
=> Observe that Narrator reads all the content correctly.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issues
 * [JDK-8087557](https://bugs.openjdk.org/browse/JDK-8087557): [Win] [Accessibility, Dialogs] Alert Dialog content is not fully read by Screen Reader
 * [JDK-8292658](https://bugs.openjdk.org/browse/JDK-8292658): [Win] [Accessibility, Dialogs] Alert Dialog content is not fully read by Screen Reader (**CSR**)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/873/head:pull/873` \
`$ git checkout pull/873`

Update a local copy of the PR: \
`$ git checkout pull/873` \
`$ git pull https://git.openjdk.org/jfx pull/873/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 873`

View PR using the GUI difftool: \
`$ git pr show -t 873`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/873.diff">https://git.openjdk.org/jfx/pull/873.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
This will need a CSR since you are adding a new enum value to `AccessibleRole`.

/csr
/reviewers 2

File: modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java [hunk 1]
```diff
@@ -822,4 +822,16 @@ public enum AccessibleRole {
...
      */
     TREE_VIEW,
+
+    /**
+     * Dialog role.
+     * <p>
+     * Attributes:
+     * <ul>
+     * <li> {@link AccessibleAttribute#TEXT} </li>
+     * <li> {@link AccessibleAttribute#ROLE_DESCRIPTION} </li>
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java:832]: What is the purpose of adding `ROLE_DESCRIPTION` here? It isn't listed anywhere else except as an optional `Node` attribute.

File: modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java [hunk 2]
```diff
@@ -822,4 +822,16 @@ public enum AccessibleRole {
...
+     * <li> {@link AccessibleAttribute#CHILDREN} </li>
+     * </ul>
+     */
+    DIALOG
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java:836]: You need to add a `@since 20` tag.

**Review by kevinrushforth — CHANGES_REQUESTED** [review]
I left some comments on the `AccessibleRole.DIALOG` docs. Also, I get the following exception on macOS with VoiceOver when opening any dialog:

```
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException: Cannot read field "ptr" because "macRole" is null
	at javafx.graphics@20-internal/com.sun.glass.ui.mac.MacAccessible.accessibilityAttributeValue(MacAccessible.java:1305)
	at javafx.graphics@20-internal/com.sun.glass.ui.mac.MacApplication._enterNestedEventLoopImpl(Native Method)
	at javafx.graphics@20-internal/com.sun.glass.ui.mac.MacApplication._enterNestedEventLoop(MacApplication.java:168)
	at javafx.graphics@20-internal/com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:515)
	at javafx.graphics@20-internal/com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
	at javafx.graphics@20-internal/com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:631)
	at javafx.graphics@20-internal/javafx.stage.Stage.showAndWait(Stage.java:469)
	at javafx.controls@20-internal/javafx.scene.control.HeavyweightDialog.showAndWait(HeavyweightDialog.java:162)
	at javafx.controls@20-internal/javafx.scene.control.Dialog.showAndWait(Dialog.java:345)
	at hello.HelloAlert.lambda$start$0(HelloAlert.java:79)
...
```

You will need to at least implement a stub on macOS.

File: modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java (hunk 1 above)
Comment (arapte) [reply modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java:832]: Reason to include `ROLE_DESCRIPTION` explicitly is that role description is fetched and read by the accessibility client application.
The change in MacAccessible.java line 1302 and in WinAccessible line 799, are specifically for ROLE_DESCRIPTION.
Windows Narrator appends this role description after dialog title.

File: modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java (hunk 2 above)
Comment (arapte) [reply modules/javafx.graphics/src/main/java/javafx/scene/AccessibleRole.java:838]: Added the tag.

**Comment by arapte** [comment]
> I get the following exception on macOS with VoiceOver when opening any dialog:

It is fixed in next [commit](https://github.com/openjdk/jfx/pull/873/commits/39d1896751fe6709469ff79380ad15e7493f4879) 

It was caused when VoiceOver requested role description for DIALOG role.

**Review by kevinrushforth — APPROVED** [review]
Looks good now.

**Review by aghaisas — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**